x/pairing/keeper/pairing.go

```diff
@@ -152,15 +154,29 @@ func (k Keeper) getPairingForClient(ctx sdk.Context, chainID string, clientAddre
 		return nil, 0, false, fmt.Errorf("did not find providers for pairing: epoch:%d, chainID: %s", block, chainID)
 	}
 
-	providers, err = k.calculatePairingForClient(ctx, possibleProviders, projectToPair, block, chainID, geolocation, epochHash, providersToPair)
+	selectedProvidersFlag := false
+	switch selectedProvidersMode {
+	case projectstypes.Policy_EXCLUSIVE, projectstypes.Policy_MIXED:
+		possibleProviders, err = k.getStakeEntriesOfSelectedProviders(possibleProviders, selectedProvidersList)
+		if err != nil {
+			return nil, 0, false, err
+		}
+
+		if uint64(len(possibleProviders)) < providersToPair {
+			providersToPair = uint64(len(possibleProviders))
```

Comment:
if this is true we dont need to calculate pairing. or calculate pairing for mixed?

Reply:
you're right. I continued with the usual pairing to remove frozen providers, but it was unnecessary. In mixed more, I'd have to consider the providersToPair variable